### DISCUSSION TITLE
[14일차-예진] use strict가 무엇인지, 장단점

### DISCUSSION THREAD

**Discussion by ez1n** [zzangCS/zzangInterview#48] [프론트 면접 질문]
use strict가 무엇이고 사용했을 때 장단점에 대해 설명해주세요.

**Comment by ez1n** [comment]
**use strict**
자바스크립트 언어의 문법을 보다 엄격히 적용하여 기존에는 무시되던 오류를 발생시킬 가능성이 높거나 자바스크립트 엔진의 최적화 작업에 문제를 일으킬 수 있는 코드에 대해 명시적인 에러를 발생시키는 것
⇒ 전역에서 선언하는 경우 모든 소스코드가 대상이 되며 로컬에서 선언하는 경우 함수 안으로 범위가 제한된다.
⇒ 삭제가 불가능한 프로퍼티를 삭제하거나 함수의 매개변수를 중복해서 사용하는 것 등이 금지된다.
⇒ 디버깅이 쉬워지고 실수로 전역변수를 생성하는 것이 불가능

**단점** : use strict 모드를 지원하지 않는 브라우저의 경우 다른 방식으로 동작할 수 있다.

**Reply by ez1n** [reply]
typescript와의 차이는 무엇일까?

**Comment by skyhanull** [comment]
- 자바스크립트 언어의 문법을 보다 엄격히 적용하여 기존에는 무시되던 오류를 발생시킬 가능성이 높거나 자바스크립트 엔진의 최적화 작업에 문제를 일으킬 수 있는 코드에 대해 명시적인 에러를 발생시킨다.
 전역에서 선언하면 모든 소스코드가 대상이고, 로컬로 선언하면 함수 내에서만 대상이 된다. use strict 모드에서는 삭제가 불가능한 프로퍼티를 삭제하거나 함수의 매개변수를 중복해서 사용하는 것 등이 금지된다. 
 코드의 문제를 더 빨리 알게되기에 디버깅이 쉬워지고 실수로 전역변수를 만드는 것이 불가능하지만, 단점으로는 엄격모드를 지원하지 않는 브라우저에서는 엄격 모드의 코드가 다른 방식으로 동작할 수 있다는 것이다.
